packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -86,4 +134,131 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
             order_id: orderID,
         });
     }
+
+    private async _onHostedMethodApprove(data: ApproveDataOptions, actions: ApproveActions) {
+        try {
+            const orderDetails = await actions.order.get();
+            const consignments = this._store.getState().consignments.getConsignmentsOrThrow();
+            const lineItems = this._getLineItems();
+            if (!this._paymentMethod?.id) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+            const methodId = this._paymentMethod?.id;
+
+            if (this._currentShippingAddress) {
+                const shippingAddress = {
+                    ...this._currentShippingAddress,
+                    firstName: orderDetails.payer.name.given_name,
+                    lastName: orderDetails.payer.name.surname,
+                    email: orderDetails.payer.email_address,
+                    address1: orderDetails.purchase_units[0].shipping.address.address_line_1,
+                };
+                const consignment = {
+                    id: consignments[0].id,
+                    shippingAddress,
+                    lineItems,
+                };
+                await this._store.dispatch(this._billingAddressActionCreator.updateAddress(shippingAddress));
+
+                await this._store.dispatch(this._consignmentActionCreator.updateConsignment(consignment));
+
+                const submitOrderPayload = {};
+                const submitOrderOptions = {
+                    params: {
+                        methodId,
+                    },
+                };
+
+                await this._store.dispatch(this._orderActionCreator.submitOrder(submitOrderPayload, submitOrderOptions));
+
+                const paymentData = {
+                    formattedPayload: {
+                        vault_payment_instrument: null,
+                        set_as_default_stored_instrument: null,
+                        device_info: null,
+                        method_id: methodId,
+                        paypal_account: {
+                            order_id: data.orderID,
+                        },
+                    },
+                };
+                await this._store.dispatch(this._paymentActionCreator.submitPayment({ methodId, paymentData }));
+                window.location.assign('/checkout/order-confirmation');
```

Comment:
Instead of this can we use do something like this in https://github.com/bigcommerce/checkout-sdk-js/blob/master/packages/core/src/checkout-buttons/strategies/apple-pay/apple-pay-button-strategy.ts#L66. Since in the above approach it causes a snowflake and is not configurable.

Reply:
fixed